### DISCUSSION TITLE
Providing inputs in DI 🎁 - justangular.com

### DISCUSSION THREAD

**Discussion by giscus** [eneajaho/just-angular#18] [General]
# Providing inputs in DI 🎁 - justangular.com

Learn how to use the new signal inputs for removing prop drilling of inputs in Angular.

https://justangular.com/blog/providing-inputs-in-di/

**Comment by AmirAlOmari** [comment]
Though technically possible, IMO it breaks the parent-chile relationship as (most?) code base enforce.

Anyway, good to see Angular gaining more incredible options

**Reply by eneajaho** [reply]
Yeah true, but in complex apps, that relationship is broken from time to time from what I've seen. 

The best part of the solution I provided is that is easy to test too, but also easy to refactor to something else in the future.

**Comment by mohanramphp** [comment]
The ideology behind Component based building it to keep it compose and work in a flexible way. This kind of introduction will lead to same bottle neck we face with inheritance.

Beauty of composition is in certain situation A can be parent for B and in some other B can be parent of A  I.e A(B()) or B(A())

I feel this will break the composition and forcefully instruct consumer to keep the parent child order defined correctly.

**Reply by eneajaho** [reply]
The pattern I've showcased is mostly to skip prop-drilling (passing data over and over again using inputs). Of course everything should be used with caution (meaning devs of the project should understand what it's happening in this case)

**Comment by fidgett** [comment]
Hi!

I didn't make it. It gives me an error:

TypeError: Cannot read properties of undefined (reading 'hasOwnProperty')

Example:
https://stackblitz.com/edit/stackblitz-starters-gttwko?file=src%2Findex.html

**Reply by eneajaho** [reply]
```diff
export class HeaderMenuComponent {
-  public menu = Inject(MENU);
+ public menu = inject(MENU);
}
```

**Reply by fidgett** [reply]
Oops. Thanks )
